test/new-e2e/tests/process/fargate_test.go

```diff
@@ -0,0 +1,121 @@
+	return func(ctx *pulumi.Context, env *fargateCPUStressEnv) error {
+		awsEnv, err := aws.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+
+		params := ecs.GetProvisionerParams(
+			ecs.WithAwsEnv(&awsEnv),
+			ecs.WithECSFargateCapacityProvider(),
+			ecs.WithFargateWorkloadApp(func(e aws.Environment, clusterArn pulumi.StringInput, apiKeySSMParamName pulumi.StringInput, fakeIntake *fakeintakeComp.Fakeintake) (*ecsComp.Workload, error) {
+				return cpustress.FargateAppDefinition(e, clusterArn, apiKeySSMParamName, fakeIntake)
+			}),
+		)
+
+		if err := ecs.Run(ctx, &env.ECS, params); err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func TestECSFargateTestSuite(t *testing.T) {
+	t.Parallel()
+	s := ECSFargateSuite{}
+	e2eParams := []e2e.SuiteOption{e2e.WithProvisioner(
+		e2e.NewTypedPulumiProvisioner("ecsFargateCPUStress", ecsFargateCPUStressProvisioner(), nil),
+	),
+	}
+
+	e2e.Run(t, &s, e2eParams...)
+}
+
+func (s *ECSFargateSuite) TestProcessCheck() {
+	t := s.T()
+	// PROCS-4219
+	flake.Mark(t)
+
+	// Flush fake intake to remove any payloads which may have
+	s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+
+	var payloads []*aggregator.ProcessPayload
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		var err error
+		payloads, err = s.Env().FakeIntake.Client().GetProcesses()
+		assert.NoError(c, err, "failed to get process payloads from fakeintake")
+
+		// Wait for two payloads, as processes must be detected in two check runs to be returned
+		assert.GreaterOrEqual(c, len(payloads), 2, "fewer than 2 payloads returned")
+	}, 5*time.Minute, 10*time.Second)
+
+	assertProcessCollected(t, payloads, false, "stress-ng-cpu [run]")
+	assertContainersCollected(t, payloads, []string{"stress-ng"})
+}
+
+func (s *ECSFargateSuite) TestProcessCheckInCoreAgent() {
+	t := s.T()
+	// PROCS-4219
+	flake.Mark(t)
+
+	extraConfig := runner.ConfigMap{
+		"ddagent:extraEnvVars": auto.ConfigValue{Value: "DD_PROCESS_CONFIG_RUN_IN_CORE_AGENT_ENABLED=true"},
+	}
+
+	s.UpdateEnv(e2e.NewTypedPulumiProvisioner("ecsFargateCPUStress", ecsFargateCPUStressProvisioner(), extraConfig))
+
+	// Flush fake intake to remove any payloads which may have
+	s.Env().FakeIntake.Client().FlushServerAndResetAggregators()
+
+	var payloads []*aggregator.ProcessPayload
+	assert.EventuallyWithT(t, func(c *assert.CollectT) {
+		var err error
+		payloads, err = s.Env().FakeIntake.Client().GetProcesses()
+		assert.NoError(c, err, "failed to get process payloads from fakeintake")
+
+		// Wait for two payloads, as processes must be detected in two check runs to be returned
+		assert.GreaterOrEqual(c, len(payloads), 2, "fewer than 2 payloads returned")
+	}, 2*time.Minute, 10*time.Second)
+
+	assertProcessCollected(t, payloads, false, "stress-ng-cpu [run]")
+	requireProcessNotCollected(t, payloads, "process-agent")
+	assertContainersCollected(t, payloads, []string{"stress-ng"})
+}
```

Comment:
Were we supposed to have a test for containers only, or is that in a separate PR? In https://github.com/DataDog/test-infra-definitions/pull/1181 I enabled process collection by default on Fargate for the agent in tests, but it seems like if we want to have a test for container collection only we'd need to support tweaking the agent env vars to enable container collection but not process collection

Reply:
I think we were only going to have a seperate test for manual container checks but that is not supported for ECS yet. However, I also do think it is fair to test container collection with the process check as that is part of the functionality. We have been doing this for all other environments.